src/components/Datatable/components/ColumnsControls/hooks/useColumnVisibility.ts

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from 'react';
+import { without } from 'ramda';
+import { isEmptyArray } from 'ramda-adjunct';
+
+import { DatatableStore } from '../../../Datatable.store';
+
+export const useColumnVisibility = (): {
+  hiddenColumns: string[];
+  setHiddenColumn: (id: string, value: boolean) => void;
+  storeVisibleColumns: () => void;
+  reinitializeVisibleColumns: () => void;
+  resetVisisbleColumns: () => void;
+  isInDefaultVisibility: boolean;
+} => {
+  const { hiddenColumns } = DatatableStore.useState((s) => ({
+    hiddenColumns: s.hiddenColumns,
+  }));
+  const [localHiddenColumns, setLocalHiddenColumns] = useState(hiddenColumns);
+
+  useEffect(() => {
+    const unsubscribe = DatatableStore.subscribe(
+      (s) => ({ hidden: s.hiddenColumns }),
+      ({ hidden }) => {
+        setLocalHiddenColumns(hidden);
+      },
+    );
+
+    return () => {
+      unsubscribe();
+    };
+  }, []);
+  const setHiddenColumn = (id, value) => {
+    if (value === true) {
+      setLocalHiddenColumns((prev) => without([id], prev));
+    } else {
+      setLocalHiddenColumns((prev) => [...prev, id]);
```

Comment:
Not sure what our current stance on ramda is but I'd either use a filter instead of `without` or use a proper functional expression to create the array. Just want to avoid using ramda as a toolkit library. But if we're okay with this, no arguments here.

Reply:
Ramda is 🔥  🤣  I'm fine with using `filter` in imperative code here. I used `without` because it was shorter 😄